server/json_client.go

```diff
@@ -0,0 +1,105 @@
+package server
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/mysterium/node/identity"
+	"net/http"
+	"net/url"
+)
+
+const (
+	mysteriumAgentName       = "goclient-v0.1"
+	authenticationHeaderName = "Authorization"
+	authenticationSchemaName = "Signature"
+)
+
+var noSignature = identity.SignatureBytes(nil)
+
+type HttpClient interface {
```

Comment:
It could be event simpler interface: `httpClient.Do(request)` which is compatible with `http.Client`

Reply:
native httpClient.Post get etc hides constructed request objects so there is no way insert custom headers, access bodies etc. httpClient.Do is too low level - operates on req created elsewere. The point of this interface was to provide more flexible http actions to uper level component (mysterium_api_client) and to have reusable bit of functionality - http client with request bodies serialized to json and auto signed if needed. It could disappear after we fully migrate mysterium api to signed requests.